app/controllers/days_controller.rb

```diff
@@ -85,13 +79,13 @@ def destroy
 
     def set_day
 
-      @day = Day.find(params[:id])
+      @day = current_user.days.find(params[:id])
 
     end
 
     def day_params
 
-      params.require(:day).permit(:mooddate, :moodrating, :moodjournal, :moodword, :isbookmarked)
+      params.require(:day).permit(:current_user, :mooddate, :moodrating, :moodjournal, :moodword, :isbookmarked)
```

Comment:
the day is associated with the currently logged-in user without explicitly passing the user ID in the form parameters so you can remove :current_user from the params.